shared_model/interfaces/queries/get_account_asset_transactions.hpp

```diff
@@ -45,12 +45,16 @@ namespace shared_model {
       /**
        * @return pager of requested transactions
        */
-      virtual const interface::Pager &pager() const = 0;
+      virtual const Pager &pager() const = 0;
 
       OldModelType *makeOldModel() const override {
         auto oldModel = new iroha::model::GetAccountAssetTransactions;
         oldModel->account_id = accountId();
         oldModel->assets_id = assetsId();
+        // placement-new is used to avoid from allocating old style pager twice
+        auto oldModelPager =
```

Comment:
extract for template function maybe? e.g in Pager:
```cpp
template<typename T>
static void init(T obj) {
  auto oldModelPager = std::unique_ptr<iroha::model::Pager>(obj.pager().makeOldModel());
  new (&oldModel->pager) iroha::model::Pager(*oldModelPager);
}
```
Also would be nice to add `pager()` in `Query` class so remove template param and pass just exact type

Reply:
I wrote following code in Pager class
```cpp
      template <typename T, typename OldModelQueryType>
      DEPRECATED static void initAllocatedOldModel(T obj, OldModelQueryType *oldModel) {
        auto oldModelPager =
            std::unique_ptr<iroha::model::Pager>(obj.pager().makeOldModel());
        new (&oldModel->pager) iroha::model::Pager(*oldModelPager);
      }
```
But compilation error occurs because 
```
/opt/iroha/shared_model/interfaces/common_objects/pager.hpp:78:34: error: invalid abstract parameter type 'shared_model::interface::GetAccountAssetTransactions'
In file included from /opt/iroha/shared_model/interfaces/queries/query.hpp:27:0,
                 from /opt/iroha/shared_model/interfaces/impl.cpp:23:
/opt/iroha/shared_model/interfaces/queries/get_account_asset_transactions.hpp:33:11: note:   because the following virtual functions are pure within 'shared_model::interface::GetAccountAssetTransactions':
     class GetAccountAssetTransactions
           ^
```

Reply:
Seems that passing `T` by ref may help